src/main/java/com/exadel/project/administrator/service/AdministratorService.java

```diff
@@ -72,10 +72,10 @@ public AdministratorDto getById(Long id) throws EntityNotFoundException {
     }
 
     public AdministratorDto addAdministrator(AdministratorDto administratorDto) throws EntityAlreadyExistsException {
-        if (administratorRepository.findAdministratorByEmail(administratorDto.getEmail()) != null) {
+        if (administratorRepository.findAdministratorByEmail(administratorDto.getEmail()) != null
+                || administratorRepository.findAdministratorByLogin(administratorDto.getLogin()) != null) {
```

Comment:
Let's move all validations in separate spring component AdministratorValidator

Reply:
thanks, fixed it